src/options.c

```diff
@@ -433,6 +431,18 @@ void optionsParseThumbnail(char* optarg)
     }
 }
 
+void optionsParseFileName(char* fileName)
+{
+    opt.outputFile = fileName;
+
+    if (strlen(opt.outputFile) > MAX_OUTPUT_FILENAME)
+        errx(EXIT_FAILURE,"output filename too long, must be "
+            "less than %d characters", MAX_OUTPUT_FILENAME);
+
+    if (opt.thumb)
```

Comment:
This only worked (note the past) when the filename was the last parameter.

This way it will fail: `scrot -F file.png -t 50%`

Because the variable `opt.thumbFile`  will be unassigned.
The next patch could fix the problem. If you agree, implement it yourself.

```patch
diff --git a/src/options.c b/src/options.c
index c0feada..6426567 100644
--- a/src/options.c
+++ b/src/options.c
@@ -339,6 +339,9 @@ void optionsParse(int argc, char** argv)
             warnx("unrecognised option %s", argv[optind++]);
     }
 
+    if (opt.thumb && opt.outputFile)
+        opt.thumbFile = nameThumbnail(opt.outputFile);
+
     /* So that we can safely be called again */
     optind = 1;
 }
@@ -438,9 +441,6 @@ void optionsParseFileName(char* fileName)
     if (strlen(opt.outputFile) > MAX_OUTPUT_FILENAME)
         errx(EXIT_FAILURE,"output filename too long, must be "
             "less than %d characters", MAX_OUTPUT_FILENAME);
-
-    if (opt.thumb)
-        opt.thumbFile = nameThumbnail(opt.outputFile);
 }
 
 void optionsParseNote(char* optarg)
``

Reply:
Thank you! I'd look into this.